src/blockdev.rs

```diff
@@ -970,10 +970,42 @@ pub fn detect_formatted_sector_size(buf: &[u8]) -> Option<NonZeroU32> {
 }
 
 /// Checks if underlying device is IBM DASD disk
-pub fn is_dasd(device: &str) -> Result<bool> {
+pub fn is_dasd(device: &str, fd: Option<&mut File>) -> Result<bool> {
     let target =
         canonicalize(device).with_context(|| format!("getting absolute path to {}", device))?;
-    Ok(target.to_string_lossy().starts_with("/dev/dasd"))
+    if target.starts_with("/dev/dasd") {
```

Comment:
Do these checks work?  `PathBuf.starts_with()` is [documented](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.starts_with) to compare only whole path components.

Reply:
missed that part, fixed